osu.Game.Rulesets.Osu/Difficulty/Skills/Speed.cs

```diff
@@ -42,7 +44,21 @@ protected override double StrainValueOf(DifficultyHitObject current)
             var osuCurrent = (OsuDifficultyHitObject)current;
 
             double distance = Math.Min(single_spacing_threshold, osuCurrent.TravelDistance + osuCurrent.JumpDistance);
-            double deltaTime = Math.Max(max_speed_bonus, current.DeltaTime);
+            double deltaTime = current.DeltaTime;
+
+            // Aim to nerf cheesy rhythms (Very fast consecutive doubles with large deltatimes between)
+            double deltaTimeThreshold = greatWindow * 2;
+
+            if (Previous.Count > 0 && deltaTime < deltaTimeThreshold && Previous[0].DeltaTime > deltaTime)
+            {
+                double speedWindowRatio = deltaTime / deltaTimeThreshold;
+                deltaTime = Interpolation.Lerp(Previous[0].DeltaTime, deltaTime, speedWindowRatio);
+            }
+
+            // Cap deltatime to the OD 300 hitwindow.
+            // 0.77 is derived from making sure 260bpm OD8 streams aren't nerfed 
+            var hitWindowNerf = deltaTime / (greatWindow * 2 * 0.77);
```

Comment:
How does this work? I tried to double-check this and got something completely different:

```
> (1 minute / 260 / 4) to ms

  ((1 * minute) / 260) / 4 = approx. 57.692308 ms

> 2 * (79 - (8 * 6) + 0.5) ms * 0.77

  2 * ((79 - (8 * 6) + 0.5) * millisecond) * 0.77 = 48.51 ms
```

Is my assumption that those streams in question are 1/4 260bpm streams wrong?

Reply:
This is actually meant to be '260bpm streams above OD7 aren't nerfed'

Reply:
Ah, that's on me not being exact with my comment - I meant to write "aren't nerfed _harshly_", whoops

Reply:
Yep, that checks out, thanks for clarifying.

```
> 2 * ((79 - (7 * 6) + 0.5) ms) * 0.77

  2 * ((79 - (7 * 6) + 0.5) * millisecond) * 0.77 = 57.75 ms
```